### DISCUSSION TITLE
Find tmxlite on Windows

### DISCUSSION THREAD

**Discussion by LuckeeDev** [fallahn/tmxlite#144] [Q&A]
I'm trying to build a program that uses `tmxlite` on Windows. I built the library and everything seems to work fine, how can I allow CMake to find it? Here's my directory tree:
```tree
...
├── CMakeLists.txt
└── modules
    └── tmxlite
        ├── bin
        │   └── libtmxlite.dll
        ├── include
        │   └── tmxlite
        │       ├── Config.hpp
        │       ├── FreeFuncs.hpp
        │       ├── ImageLayer.hpp
        │       ├── Layer.hpp
        │       ├── LayerGroup.hpp
        │       ├── Map.hpp
        │       ├── Object.hpp
        │       ├── ObjectGroup.hpp
        │       ├── ObjectTypes.hpp
        │       ├── Property.hpp
        │       ├── TileLayer.hpp
        │       ├── Tileset.hpp
        │       ├── Types.hpp
        │       ├── Types.inl
        │       └── detail
        │           ├── Android.hpp
        │           └── Log.hpp
        └── lib
            ├── libtmxlite.dll.a
            └── pkgconfig
                └── tmxlite.pc
...
```
I'm having troubles with both header files and library files. It can't find neither of them.

**Comment by fallahn** [comment]
If you've installed the library to a sub-directory of your project, you can add it to your project's cmake file directly:

    include_directories(${CMAKE_SOURCE_DIR}/modules/tmxlite/include)
    link_directories(${CMAKE_SOURCE_DIR}/modelues/tmxlite/lib)

and then add it to the linker list of your executable

      add_executable(foo ${FOO_SRCS})
      target_link_libraries(foo tmxlite)

**Comment by LuckeeDev** [comment]
Thank you, that worked! And what should I do if I installed `tmxlite` to a global folder?

**Comment by fallahn** [comment]
It's not so straight forward on windows, as there aren't default library paths like on linux/unix. This Stack Overflow answer might help: https://stackoverflow.com/questions/21314893/what-is-the-default-search-path-for-find-package-in-windows-using-cmake

**Comment by LuckeeDev** [comment]
Thank you!
